R/volcano_server.R

```diff
@@ -38,15 +38,54 @@ volcano_server <- function(input, output, session, params) {
             stat=input$calculation_type
         )
     })
-        
+
+    groups <- reactive({
+        c(
+            mapping()$comparison_group, 
+            mapping()$reference_group
+        )
+    })
+
     # draw the chart
     output$volcanoPlot <- renderPlotly({
         volcanoPlot(
             stats(), 
-            GroupLabels = c(
-                mapping()$comparison_group, 
-                mapping()$reference_group
-            )
+            GroupLabels = groups()
         )
     })
+
+    stats_sub <- reactive({
+        plotly_d <- event_data("plotly_click")
+        if (is.null(plotly_d)) {
+            NULL
+        }else{
+            stats()[plotly_d$pointNumber+1,]
+        }
+    })
+
+    ae_sub <- reactive({
+        all<-params()$data$aes
+        # subset to selected strata
+        sub1<-all[all[params()$settings$aes$bodsys_col]==stats_sub()$strata,]
+
+        # subset to selected groups
+        #dm <- params()$data$dm
+        #ids <- dm[dm[params()$settings$dm$treatment_col] %in% groups(),]%>%
+        #    pull(params()$settings$dm$id_col)
+        #print(ids)
+        #sub2<-sub1[sub1[params()$settings$dm$id_col,] %in% ids,]
+
+        sub1
+    })
+
+    output$click <- renderText({
```

Comment:
I'm not sure why, but at least on my computer this sentence can't not go longer than the width of the plot.

![image](https://user-images.githubusercontent.com/22552876/185637784-be70e656-3504-441a-b8b2-35453082612c.png)

![image](https://user-images.githubusercontent.com/22552876/185637886-33101c6e-1a8f-4af6-812f-5d0d35f916eb.png)